python-frontend/src/test/java/org/sonar/python/tree/CallExpressionImplTest.java

```diff
@@ -102,4 +111,95 @@ public void not_callable_callee_symbol_type() {
   public void null_callee_symbol_type() {
     assertThat(lastExpression("x()").type()).isEqualTo(InferredTypes.anyType());
   }
+
+  @Test
+  public void method_call_with_declared_type() {
+    assertThat(lastExpression(
+      "def foo(param: str):",
+      "  x = param.capitalize()",
+      "  x"
+    ).type()).isEqualTo(DECL_STR);
+
+    assertThat(lastExpression(
+      "def foo(param: A):",
+      "  x = param.capitalize()",
+      "  x"
+    ).type()).isEqualTo(anyType());
+
+    assertThat(lastExpression(
+      "def foo(param):",
+      "  x = param.capitalize()",
+      "  x"
+    ).type()).isEqualTo(anyType());
+
+    // TODO: handle member access dependencies for declared type
```

Comment:
What would be required for this? I assume we'd need to link the last `capitalize` called to the declared type of `param.capitalize`? Do we have a ticket for that?

Reply:
done: https://jira.sonarsource.com/browse/SONARPY-781